agent/testflinger_agent/job.py

```diff
@@ -126,7 +126,7 @@ def run_test_phase(self, phase, rundir):
             )
         if phase == "allocate":
             self.allocate_phase(rundir)
-        return exitcode
+        return exitcode, exit_event, exit_reason
```

Comment:
exit_event and exit_reason are the event_name and detail_msg fields respectively. They get used in agent.py when the job gets run.

Reply:
They were landed as part of a previous PR to make it easier to find this information, so that it could be returned when the phase finishes running. This way we know the reason it stopped and can send that up to the server to be logged, stored, or forwarded to test-observer as needed from one place